src/qibolab/transpilers/general_connectivity.py

```diff
@@ -37,25 +41,71 @@ class Transpiler:
         _added_swaps (int): number of swaps added to the circuit to match connectivity.
     """
 
-    def __init__(self, connectivity, init_method="greedy", init_samples=None):
+    def __init__(
+        self, connectivity, init_method="greedy", init_samples=None, two_qubit_natives="CZ", sampling_split=1.0
+    ):
         self.connectivity = connectivity
         self.init_method = init_method
         if self.init_method is QubitInitMethod.greedy and init_samples is None:
             init_samples = DEFAULT_INIT_SAMPLES
         self.init_samples = init_samples
+        self.two_qubit_natives = two_qubit_natives
+        self.sampling_split = sampling_split
 
         self._circuit_repr = None
         self._mapping = None
         self._graph = None
         self._qubit_map = None
+        self._transpiled_circuit = None
         self._circuit_position = 0
         self._added_swaps = 0
 
-    def transpile(self, qibo_circuit):
-        """Qubit mapping initialization and circuit transpiling.
+    def transpile(self, circuit, fuse_one_qubit=False, fusion_algorithm=False):
+        """Full transpilation, match connectivity and translation into native gates.
 
         Args:
-            qibo_circuit (qibo.Circuit): circuit to be transpiled.
+            circuit (qibo.Circuit): circuit to be transpiled.
+            fuse_one_qubit (bool): Fuse two or more one qubit gates in sequence
+            fusion_algorithm (bool): Try to reduce the number of SWAP in the transpiler by using qibo fusion algorithm
+
+        Returns:
+            transpiled_circuit (qibo.Circuit): circut mapped to hardware topology with only native gates.
+            final_mapping (dict): final qubit mapping.
+            init_mapping (dict): initial qubit mapping.
```

Comment:
Since this function is part of the interface, could you please document more what "initial" and "final" mean here and what types the dictionaries are mapping to (eg. int to int)? I guess `final_mapping` is a map of the original qubit positions to one after the transpilation, but what is the `init_mapping`?